cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/LdapResourceCRLFetcher.java

```diff
@@ -0,0 +1,223 @@
+
+    /** The connection configuration to ldap.*/
+    protected final ConnectionConfig connectionConfig;
+
+    /** Search request to find the attribute in the ldap tree.*/
+    protected final SearchRequest searchRequest;
+
+    /** The ldap provider, defaults to {@link DefaultConnectionFactory#getProvider()}. */
+    protected Provider provider = DefaultConnectionFactory.getDefaultProvider();
+
+    /** The connection factory to prep for connections. **/
+    protected ConnectionFactory connectionFactory;
+
+    private AbstractConnectionPool connectionPool;
+
+    /**
+     * Instantiates a new Ldap resource cRL fetcher.
+     * initializes the default certificate revocation attribute
+     * to be {@link #DEFAULT_CERTIFICATE_REVOCATION_LIST_ATTRIBUTE}.
+     * @param searchRequest the search request
+     * @param connectionConfig the connection config
+     */
+    public LdapResourceCRLFetcher(@NotNull final SearchRequest searchRequest,
+                                  @NotNull final ConnectionConfig connectionConfig) {
+        this(searchRequest, connectionConfig,
+                DEFAULT_CERTIFICATE_REVOCATION_LIST_ATTRIBUTE);
+    }
+
+    /**
+     * Instantiates a new Ldap resource cRL fetcher.
+     *
+     * @param searchRequest the search request
+     * @param connectionConfig the connection config
+     * @param certificateRevocationListAttributeName the certificate revocation list attribute name
+     */
+    public LdapResourceCRLFetcher(@NotNull final SearchRequest searchRequest,
+                                  @NotNull final ConnectionConfig connectionConfig,
+                                  @NotNull final String certificateRevocationListAttributeName) {
+        this.certificateRevocationListAttributeName = certificateRevocationListAttributeName;
+        this.connectionConfig = connectionConfig;
+        this.searchRequest = searchRequest;
+    }
+
+    @Override
+    protected X509CRL fetchInternal(final Object r) throws Exception {
+        if (r.toString().toLowerCase().startsWith("ldap")) {
+            return fetchCRLFromLdap(r);
+        }
+        return super.fetchInternal(r);
+    }
+
+    public void setProvider(final Provider provider) {
+        this.provider = provider;
+    }
+
+    public void setConnectionPool(final AbstractConnectionPool connectionPool) {
+        this.connectionPool = connectionPool;
+    }
+
+    /**
+     * Downloads a CRL from given LDAP url
+     *
+     * @param r the resource that is the ldap url.
+     * @return the x 509 cRL
+     * @throws Exception if connection to ldap fails, or attribute to get the revocation list is unavailable
+     */
+    protected X509CRL fetchCRLFromLdap(final Object r) throws Exception {
+        Connection connection = null;
+        try {
+            final String ldapURL = r.toString();
+            logger.debug("Fetching CRL from ldap {}", ldapURL);
+
+            prepareConnectionFactory(ldapURL);
```

Comment:
One thing we may want to consider the ability to configure connection pooling may start to become a performance burden if there are many URLs specified in the cert. We have encountered cases where there was more than one URL provided in the cert, and we needed to check them all. 

Now, while that feature does not presently exist (as this PR only looks at a single URL), I am wondering if we need to ultimately provide other impls that simply do not do any sort of pooling and immediately close connections.